src/k8s/pkg/k8sd/api/worker_access_handler.go

```diff
@@ -13,53 +13,53 @@ import (
 	"github.com/canonical/microcluster/state"
 )
 
-func (e *Endpoints) restrictWorkers(s *state.State, r *http.Request) response.Response {
+func (e *Endpoints) restrictWorkers(s state.State, r *http.Request) (bool, response.Response) {
 	snap := e.provider.Snap()
 
 	isWorker, err := snaputil.IsWorker(snap)
 	if err != nil {
-		return response.InternalError(fmt.Errorf("failed to check if node is a worker: %w", err))
+		return false, response.InternalError(fmt.Errorf("failed to check if node is a worker: %w", err))
 	}
 
 	if isWorker {
-		return response.Forbidden(fmt.Errorf("this action is restricted on workers"))
+		return false, response.Forbidden(fmt.Errorf("this action is restricted on workers"))
 	}
 
-	return response.EmptySyncResponse
+	return true, nil
 }
 
 // ValidateWorkerInfoAccessHandler access handler checks if the worker is allowed to access this endpoint with the provided token.
-func ValidateWorkerInfoAccessHandler(nodeHeaderName string, tokenHeaderName string) func(s *state.State, r *http.Request) response.Response {
-	return func(s *state.State, r *http.Request) response.Response {
+func ValidateWorkerInfoAccessHandler(nodeHeaderName string, tokenHeaderName string) func(s state.State, r *http.Request) (bool, response.Response) {
+	return func(s state.State, r *http.Request) (bool, response.Response) {
 		name := r.Header.Get(nodeHeaderName)
 		if name == "" {
-			return response.Unauthorized(fmt.Errorf("missing header %q", nodeHeaderName))
+			return false, response.Unauthorized(fmt.Errorf("missing header %q", nodeHeaderName))
 		}
 		hostname, err := utils.CleanHostname(name)
 		if err != nil {
-			return response.BadRequest(fmt.Errorf("invalid hostname %q: %w", hostname, err))
+			return false, response.BadRequest(fmt.Errorf("invalid hostname %q: %w", hostname, err))
 		}
 
 		token := r.Header.Get(tokenHeaderName)
 		if token == "" {
-			return response.Unauthorized(fmt.Errorf("invalid token"))
+			return false, response.Unauthorized(fmt.Errorf("invalid token"))
 		}
 
 		var tokenIsValid bool
-		if err := s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
+		if err := s.Database().Transaction(r.Context(), func(ctx context.Context, tx *sql.Tx) error {
 			var err error
 			tokenIsValid, err = database.CheckWorkerNodeToken(ctx, tx, hostname, token)
 			if err != nil {
 				return fmt.Errorf("failed to check worker node token: %w", err)
 			}
 			return nil
 		}); err != nil {
-			return response.InternalError(fmt.Errorf("check token database transaction failed: %w", err))
+			return false, response.InternalError(fmt.Errorf("check token database transaction failed: %w", err))
 		}
 		if !tokenIsValid {
-			return response.Unauthorized(fmt.Errorf("invalid token"))
+			return false, response.Unauthorized(fmt.Errorf("invalid token"))
 		}
 
-		return response.EmptySyncResponse
+		return true, nil
```

Comment:
and here?

Reply:
Actually, I think both work, returning `nil` and `EmptySyncResponse`. 
Just followed along with how they do it now in their default access handler: 
https://github.com/canonical/microcluster/blob/main/rest/access/handlers.go#L32